### PR TITLE
[clang-tidy][doc] Show config for all checks

### PR DESCRIPTION
**Pull request by gamesh411**
CodeChecker checkers now shows configuration options for all available
checks instead of only default-enabled ones.

### CODE REVIEW

File: analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py [hunk 1]
```diff
@@ -79,14 +79,21 @@ class ClangTidy(analyzer_base.SourceAnalyzer):
...
+        return [cfg_handler.analyzer_binary, "-checks=*"]
+
     @classmethod
     def get_analyzer_checkers(cls, cfg_handler, environ):
         """
         Return the list of the supported checkers.
         """
         try:
+            list_invocation =\
+                cls.invocation_with_all_checks(cfg_handler)\
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py:93]: I think there is no need to create a separate function for this. You can add this option to the original list.

Also could you please extend the documentation of this function a little bit more that this function will return the list of `ALL` the supported checkers?

File: analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py [hunk 2]
```diff
@@ -98,8 +105,11 @@ def get_analyzer_checkers(cls, cfg_handler, environ):
     @classmethod
     def get_checker_config(cls, cfg_handler, environ):
         try:
+            config_dump_invocation =\
+                cls.invocation_with_all_checks(cfg_handler)\
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py:109]: Same as above.

File: analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py [hunk 3]
```diff
@@ -111,8 +121,11 @@ def get_checker_config(cls, cfg_handler, environ):
     @classmethod
     def get_analyzer_config(cls, cfg_handler, environ):
         try:
+            config_dump_invocation =\
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py:124]: Same as above.

**Review by csordasmarton — CHANGES_REQUESTED** [review]
I get the following error message when I tried to list the available checkers with `CodeChecker checkers --checker-config`:

```
File "codechecker/build/CodeChecker/lib/python3/codechecker_analyzer/analyzers/clangtidy/analyzer.py", line 93, in get_analyzer_checkers
    cls.invocation_with_all_checks(cfg_handler)\
TypeError: invocation_with_all_checks() takes 1 positional argument but 2 were given
```

File: analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py (hunk 1 above)
Comment (gamesh411) [reply analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py]: Fixed, and modified accordingly.

File: analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py [hunk 4]
```diff
@@ -110,9 +113,12 @@ def get_checker_config(cls, cfg_handler, environ):
 
     @classmethod
     def get_analyzer_config(cls, cfg_handler, environ):
+        """
+        Return the analyzer configuration with all checkers enabled.
+        """
         try:
             result = subprocess.check_output(
-                [cfg_handler.analyzer_binary, "-dump-config"],
+                [cfg_handler.analyzer_binary, "-dump_config", "-checks=*"],
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py:121]:
```suggestion
                [cfg_handler.analyzer_binary, "-dump-config", "-checks=*"],
```

**Review by csordasmarton — CHANGES_REQUESTED** [review]

**Review by csordasmarton — APPROVED** [review]
LGTM!

**Review by gyorb — APPROVED** [review]

**Pull request merged**